pkg/destroy/aws/aws.go

```diff
@@ -103,16 +106,13 @@ func (o *ClusterUninstaller) Run() error {
 		return err
 	}
 
-	awsConfig := &aws.Config{Region: aws.String(o.Region)}
 	awsSession := o.Session
 	if awsSession == nil {
 		// Relying on appropriate AWS ENV vars (eg AWS_PROFILE, AWS_ACCESS_KEY_ID, etc)
-		awsSession, err = session.NewSession(awsConfig)
+		awsSession, err = session.NewSession(aws.NewConfig().WithRegion(o.Region))
```

Comment:
Not sure I'm following this. Would this scenario `awsSession == nil` ever be hit? It seems to me it should not, and this is purely defensive.

Reply:
the destroy package is used by hive as a lib, i.e aws.ClusterUninstaller, therefore these is a case where they have set the session as nil. and that's why this check was added in the first place.